Add golem framework

CarlosEpiStats opened the pull request:
Adopted golem framework:

- Moved app logic to app_server.R, app_ui.R, and run_app.R
- Changed file names to fct_ and utils_
- Moved all "import from" to R/adepro-package.R
- Fixed CMD check issues with global variables and functions
- Fixed CMD check issues with documentation

CarlosEpiStats committed:
Started migration to golem framework:
- run 01_start.R and associated changes
- renamed old vignette folder to vignettes_old to avoid conflict when creating golem project
- updated README with additional fields from golem
- updated CODE_OF_CONDUCT
- added standard .R golem files

Next steps:
- run 02_dev.R
- move app code to module files

CarlosEpiStats committed:
Updated package dependencies

CarlosEpiStats committed:
Updated to golem framework

- Moved app logic to app_server.R, app_ui.R, and run_app.R
- Changed file names to fct_ and utils_
- Moved all imports to R/adepro-package.R
- Fixed CMD check issues with global variables and functions
- Fixed CMD check issues with documentation

CarlosEpiStats committed:
Updated to golem framework

- Moved app logic to app_server.R, app_ui.R, and run_app.R
- Changed file names to fct_ and utils_
- Moved all imports to R/adepro-package.R
- Fixed CMD check issues with global variables and functions
- Fixed CMD check issues with documentation
- Ammended with new file uploads (weren't tracked in the previous commit)

CarlosEpiStats committed:
Merge branch 'add-golem-framework' of https://github.com/Bayer-Group/BIC-AdEPro into add-golem-framework

CarlosEpiStats committed:
Update dependencies

- Removed dependencies on V8 an Cairo
- Add missing dependency on dplyr::n()

CarlosEpiStats committed:
Updated README

- Changed launch_adepro() for run_app()
- Updated results for R CMD check (now, 0 notes)

humberto-garciacaballero reviewed (changes requested):
I believe we can remove the `LICENSE` file and leave the LICENSE.md file that was created with `usethis`. Also, in the `DESCRIPTION` file we can just leave GPL(>=3), that is, remove the + file.

CarlosEpiStats committed:
Remove LICENSE file

CarlosEpiStats committed:
Moved some functions from NAMESPACE to explicit syntax pkg::fct()

CarlosEpiStats committed:
Updated dependencies

- Changed all importFrom to pkg::fct() syntax
- Only keeping symbols %>%, := and .data
- Removed unused import package "shape"
- Renamed my.symbols() to my_symbols()

CarlosEpiStats committed:
Removed globalVariables

- Fixed "no visible binding..." check note from:
  - fct_adepro_slice_plot.R
  - app_server.R (with the exception of '.', that will require a bit more restructuring in the code)
  - fct_bar_chart.R (with the exception of "pull" for some reason, although it is called as dplyr::pull)
- Several other files to fix

humberto-garciacaballero reviewed (changes requested):
I have a few requests more. Here is the list:

- It seems that in [`app_server.R`](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/app_server.R#L12) a variable named `adepro_colors` is created but never used. The only reference that I have found with the same name is in `adepro_slice_plot.R` as a parameter with the same values. Thus, I think we can safely remove the declaration in `app_server.R`. Same happens with `aes` and `aes2` in lines [1595](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/app_server.R#L1595) and [1596](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/app_server.R#L1596)
- We are using this code `colors[1:length(input_var())]` which can be incorrect when input_var is empty. Instead, we should use `seq_along(input_var())`. This happens at least in  [`app_server.R`](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/app_server.R#L295) and some other places in that file. Similarly, we have another situation in which we use `1:nrow(tmp)`. We should instead use `seq_len(nrow(temp))`. This happens too in line 1676.
- In [`fct_adepro_slice_plot.R`](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L95) the and operand is missing another `&`. This happens too in lines [218](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L218), [222](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L222), [226](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L226), [249](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L95), [253](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L253), [257](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/fct_adepro_slice_plot.R#L257)
- There is also a NOTE coming from `devtools::check` about a dot '.' variable not being visible. This happens at least in [`app_server.R`](https://github.com/Bayer-Group/BIC-AdEPro/blob/d50792f784c88d9e630844a44d19c2e2ebb481d3/R/app_server.R#L1053). One possible solution so we don't get these notes is to replace that to use an explicit function syntax as follows:

```r
...
{function(x) round(mapply('/', x, N_treat) * 100, 1)}() %>%
...
```
- There are a few more notes about the same, maybe consider to address them so we leave the `devtools::check` output as clean as possible in this iteration.

There are a few more warnings that we can address but the list will be too long. Maybe we can start discussing these and continue later on with the rest.

CarlosEpiStats committed:
Replaced '.' for function(x) in app_server

- Changed code to avoid the use of '.', to avoid notes by CMD check

CarlosEpiStats committed:
Removed lintr warnings in fct_bar_chart.R

- seq_along() instead of 1:length()
- removed unused local objects

CarlosEpiStats committed:
Fixed lints and R CMD check notes

- removed unused old function fct_prepare_data.R (currently using fct_prepare_data_for_adepro.R)
- solved R CMD check notes and lint warnings for
  - fct_calculate_and_impute....R,
  - fct_count_event.R,
  - fct_order_patient.R,
  - fct_prepare_data_for_adepro.R

CarlosEpiStats committed:
Solved lints

- Changed 1:length() with seq_along()
- Removed unused objects
- Renamed initQ() with init_q() to keep name consistency

CarlosEpiStats committed:
Move data to internal

- Move adae_data and adsl_data to internal in R/sysdata.rda
- Update README

CarlosEpiStats committed:
Fix wrong variable bug

- It was dplyr::between(.data$day, ...) when it should be dplyr::between(day, ...)

CarlosEpiStats committed:
Silence warnings in anytime::anydate()

Many warnings were triggered each time filter_prepare_patient_data() runs, due to anytime::anydate() throwing a warning for each NA value, and it's expected to have NA values (on death date, for example).

Now, anytime::anydate() is wrapped in suppresWarnings() in helper functions is_convertible_to_date() and as_numeric_date()

There are still warnings arising from fct_calculate_and_impute....R and app_server.R, that should be tackled in future updates

CarlosEpiStats commented on R/app_server.R:
```diff
@@ -1241,8 +1239,7 @@ server <- shiny::shinyServer(function(input, output, session) {
     } else if (input$radiobutton_data == "Demo data") {
-      load("data/adsl_data.rda")
-      adsl <- adsl_data
+      adsl <- adepro::adsl_data
```
Fixed in 34dee84

humberto-garciacaballero reviewed (approved):
Looking good!
